lib/credit_card_sanitizer.rb

```diff
@@ -170,7 +172,22 @@ def is_tracking?(candidate, options)
   end
 
   def valid_numbers?(candidate, options)
-    LuhnChecksum.valid?(candidate.numbers) && valid_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+    LuhnChecksum.valid?(candidate.numbers) && valid_company_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+  end
+
+  def valid_context?(options)
+    return true if valid_prefix?(options[:prefix]) && valid_postfix?(options[:postfix])
+    !options[:parse_flanking]
+  end
+
+  def valid_prefix?(prefix)
+    return true if prefix.nil? || prefix.downcase.end_with?(*ACCEPTED_PREFIX)
+    !!/[^[:alnum:]]/.match(prefix[-1])
```

Comment:
I'm a little concerned with the performance of repeatedly creating a regex object just to see if one character is a character or digit. You could initialize a class level `Set` object with those characters and check to see if the set includes last character. Set uses hashes so it should O(1) 

Naive benchmarking shows that using a `Set` is twice as fast:

```ruby
irb(main):089:0> Benchmark.realtime { 10_000_000.times { /[^[:alnum:]]/.match('a') } }
=> 5.906984630000807
irb(main):096:0> Benchmark.realtime { 10_000_000.times { set.include?('a') } }
=> 2.5370712190015183
```